src/lib/es2019.array.d.ts

```diff
@@ -1,3 +1,10 @@
+type Flat<Arr, Depth extends number> = {

```

Comment:
I would name this FlatArray instead of Flat given the occurrences in Definitely Typed. This is similar to ConcatArray for use with Array.concat.